mp/src/game/shared/momentum/weapon/weapon_mom_stickybomblauncher.cpp

```diff
@@ -0,0 +1,383 @@
+                                                         QAngle *angForward)
+{
+#ifdef GAME_DLL
+    static ConVarRef cl_righthand("cl_righthand");
+#else
+    extern ConVar cl_righthand;
+#endif
+    if (!cl_righthand.GetBool())
+    {
+        vecOffset.y *= -1.0f;
+    }
+
+    Vector vecForward, vecRight, vecUp;
+    AngleVectors(pPlayer->EyeAngles(), &vecForward, &vecRight, &vecUp);
+
+    Vector vecShootPos = pPlayer->Weapon_ShootPosition();
+
+    // Estimate end point
+    Vector endPos = vecShootPos + vecForward * 2000.0f;
+
+    // Trace forward and find what's in front of us, and aim at that
+    trace_t tr;
+
+    CTraceFilterSimple filter(pPlayer, COLLISION_GROUP_NONE);
+    UTIL_TraceLine(vecShootPos, endPos, MASK_SOLID, &filter, &tr);
+
+    // Offset actual start point
+    *vecSrc = vecShootPos + (vecForward * vecOffset.x) + (vecRight * vecOffset.y) + (vecUp * vecOffset.z);
+
+    // Find angles that will get us to our desired end point
+    // Only use the trace end if it wasn't too close, which results
+    // in visually bizarre forward angles
+    if (tr.fraction > 0.1f)
+    {
+        VectorAngles(tr.endpos - *vecSrc, *angForward);
+    }
+    else
+    {
+        VectorAngles(endPos - *vecSrc, *angForward);
+    }
+}
+
+float CMomentumStickybombLauncher::GetProjectileSpeed()
+{
+    return RemapValClamped((gpGlobals->curtime - m_flChargeBeginTime), 0.0f, 4.0f, 900, 2400);
+}
+
+CBaseEntity *CMomentumStickybombLauncher::FireProjectile(CMomentumPlayer *pPlayer)
+{
+    CBaseEntity *pProjectile = FireStickybomb(pPlayer);
+    if (pProjectile)
+    {
+#ifdef GAME_DLL
+        // If we've gone over the max stickybomb count, detonate the oldest
+        if (m_Stickybombs.Count() >= MOM_WEAPON_STICKYBOMB_COUNT)
+        {
+            CMomStickybomb *pTemp = m_Stickybombs[0];
+            if (pTemp)
+            {
+                pTemp->Detonate(); // explode NOW
+            }
+
+            m_Stickybombs.Remove(0);
+        }
+
+        CMomStickybomb *pStickybomb = (CMomStickybomb *)pProjectile;
+        StickybombHandle hHandle;
+        hHandle = pStickybomb;
+        m_Stickybombs.AddToTail(hHandle);
+
+        m_iStickybombCount = m_Stickybombs.Count();
+#endif
+    }
+    return pProjectile;
+}
+
+CBaseEntity *CMomentumStickybombLauncher::FireStickybomb(CMomentumPlayer *pPlayer)
+{
+#ifdef GAME_DLL
+    static ConVarRef cl_righthand("cl_righthand");
+#else
+    extern ConVar cl_righthand;
+#endif
+    Vector vecForward, vecRight, vecUp;
+
+    if (!cl_righthand.GetBool())
+    {
+        vecRight.y *= -1.0f;
+    }
+
+#ifdef GAME_DLL
+
+    AngleVectors(pPlayer->EyeAngles(), &vecForward, &vecRight, &vecUp);
+
+    // Create grenades here!!
+    Vector vecSrc = pPlayer->Weapon_ShootPosition();
+    vecSrc += vecForward * 16.0f + vecRight * 8.0f + vecUp * -6.0f;
+
+    Vector vecVelocity = (vecForward * GetProjectileSpeed()) + (vecUp * 200.0f) + (random->RandomFloat(-10.0f, 10.0f) * vecRight) +
```

Comment:
Should there be RNG with how stickies are launched?

We tend to remove RNG from most game modes, this may require more discussion in the discord.

Reply:
I was considering it, but I wanted to have it be equal to TF2 for now and then once stickies are more fleshed out, we could have a discussion on what to do with the RNG.